### PR TITLE
Fix insertOrUpdate plugin may failed due to concurrent operation

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java [hunk 1]
```diff
@@ -51,10 +57,21 @@ public int addOrUpdatePluginDefine(PluginDefine pluginDefine) {
...
+            try {
+                if (pluginDefineMapper.insert(pluginDefine) == 1) {
+                    return pluginDefine.getId();
+                } else {
+                    throw new TaskPluginException(String.format("Failed to insert plugin definition, pluginName: %s, pluginType: %s", pluginDefine.getPluginName(), pluginDefine.getPluginType()));
+                }
+            } catch (TaskPluginException ex) {
+                throw ex;
+            } catch (Exception ex) {
+                logger.info("Insert plugin definition error, there may already exist a plugin");
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java]:
```suggestion
                logger.error("Insert plugin definition error, there may already exist a plugin", ex);
```

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java [hunk 2]
```diff
@@ -51,10 +57,21 @@ public int addOrUpdatePluginDefine(PluginDefine pluginDefine) {
...
         PluginDefine currPluginDefine = pluginDefineMapper.queryByNameAndType(pluginDefine.getPluginName(), pluginDefine.getPluginType());
         if (currPluginDefine == null) {
-            if (pluginDefineMapper.insert(pluginDefine) == 1 && pluginDefine.getId() > 0) {
-                return pluginDefine.getId();
+            try {
+                if (pluginDefineMapper.insert(pluginDefine) == 1) {
+                    return pluginDefine.getId();
+                } else {
+                    throw new TaskPluginException(String.format("Failed to insert plugin definition, pluginName: %s, pluginType: %s", pluginDefine.getPluginName(), pluginDefine.getPluginType()));
+                }
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java]:
```suggestion
                } 
                throw new TaskPluginException(String.format("Failed to insert plugin definition, pluginName: %s, pluginType: %s", pluginDefine.getPluginName(), pluginDefine.getPluginType()));
```

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java]: Done.

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/PluginDao.java]: Done.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**